instrumentation/net/http/otelhttp/config.go

```diff
@@ -59,18 +59,17 @@ func (o optionFunc) apply(c *config) {
 // newConfig creates a new config struct and applies opts to it.
 func newConfig(opts ...Option) *config {
 	c := &config{
-		Propagators:    otel.GetTextMapPropagator(),
-		TracerProvider: otel.GetTracerProvider(),
-		MeterProvider:  global.GetMeterProvider(),
+		Propagators:   otel.GetTextMapPropagator(),
+		MeterProvider: global.GetMeterProvider(),
 	}
 	for _, opt := range opts {
 		opt.apply(c)
 	}
 
-	c.Tracer = c.TracerProvider.Tracer(
-		instrumentationName,
-		trace.WithInstrumentationVersion(contrib.SemVersion()),
-	)
+	// Tracer is only initialized if manually specified. Otherwise, can be passed with the tracing context.
```

Comment:
a comment in code like this is not enough for documenting the behavior
it should be at least noted in "entry" point functions/types e.g. `NewHandler` and `NewTransport`

Moreover the new feature  should be noted in `CHANGELOG.md`

Reply:
This is not documenting behavior but explaining why the `.Meter` and `.Tracer` are handled differently to the reader. This switch could be in `RoundTrip()` as well.

Reply:
Added changelog item to the "Added" section. Could have been in "Fixed" as well as if the global tracer and the parent span tracer would not have matched previously I think it would have lead to some bad behavior. Lmk if you want to change that.

Reply:
For sure it should not be in "fixed":

> particularly while the question of even getting a `TracerProvider` from a `Span` is still outstanding in the spec.

Therefore if we accept such behavior as acceptable, then, in my opinion, it should be explicitly described in the godoc.